libpldmresponder/fru.cpp

```diff
@@ -918,33 +926,40 @@ void FruImpl::processFruPresenceChange(const DbusChangedProps& chProperties,
     static constexpr auto portInterface =
         "xyz.openbmc_project.Inventory.Item.Connector";
 
-#ifdef OEM_IBM
-    if (fruInterface == "xyz.openbmc_project.Inventory.Item.PCIeDevice")
-    {
-        if (!pldm::responder::utils::checkFruPresence(fruObjPath.c_str()))
-        {
-            return;
-        }
-        pldm::responder::utils::findPortObjects(fruObjPath, portObjects);
-        /*   std::cout << "after finding the port objects " <<
-           portObjects.size()
-                     << std::endl;*/
-    }
-    // as per current code the ports do not have Present property
-#endif
 
     // if(fruInterface != "xyz.openbmc_project.Inventory.Item.PCIeDevice")
     {
         if (newPropVal)
         {
+            info("Presence changed to true");
             buildIndividualFRU(fruInterface, fruObjPath);
-            for (auto portObject : portObjects)
+            if((fruInterface == "xyz.openbmc_project.Inventory.Item.PCIeDevice") && (checkIfIBMCableCard(fruObjPath)))
```

Comment:
should this not be` !(checkIfIBMCableCard(fruObjPath)` ?

Reply:
checkIfIBMCableCard function returns true if the adapter is IBM Cable card. And we expect ports under cable card. 
https://github.com/ibm-openbmc/pldm/blob/391c4012a85fb85d5b9d9a37acd63f049d9616fc/oem/ibm/libpldmresponder/utils.cpp#L427